moviepy/video/tools/drawing.py

```diff
@@ -219,27 +219,26 @@ def color_split(size,x=None,y=None,p1=None,p2=None,vector=None,
     >>> color_split(size, x=50, col1=[255,0,0], col2=[0,255,0])
     >>> # An image splitted along an arbitrary line (see below) 
     >>> color_split(size, p1=[20,50], p2=[25,70] col1=0, col2=1)
-        
+            
     """
     
     if grad_width or ( (x is None) and (y is None)):
-        if p2 is not None:
+        if p2:
             vector = (np.array(p2) - np.array(p1))
-        elif x is not None:
+        elif x:
             vector = np.array([0,-1.0])
             p1 = np.array([x, 0])
-        elif y is not None:
+        elif y:
```

Comment:
Not sure about this one. Under your change, if only one of x and y is passed in (which is the intended way of doing it), _but_ it is passed in as 0, then it will crash on line 235 with a `TypeError: cannot unpack non-iterable NoneType object` because `vector` will not be set.

Under the original version, vector and p1 get set and it will _probably_ work fine, setting it all to one colour, which, though useless, I think is preferable.